profile-specific-challenges/frontend/components/idioms.jsx

```diff
@@ -410,16 +382,49 @@ export function RenderHookComponent() {
 }
 
 // Prop Drilling
+export function ExcessivePropDrilling() {
+  const counter = useState(0);
+  const [count, setCount] = counter;
 
-function Child3({ counter } = { counter: number }) {
-  <div>{counter}</div>;
+  return (
+    <div>
+      <p>Count: {count}</p>
+      <button onClick={() => setCount((o) => o + 1)}>Increment</button>
+
+      <Multiplications counter={counter} />
+      <Divisions counter={counter} />
+    </div>
+  );
 }
-function Child2({ counter } = { counter: number }) {
-  <Child3 counter={counter} />;
+
+function MultBy5({ setCount }) {
+  return (
+    <button onClick={() => setCount((old) => old * 5)}>Multiply by 5</button>
+  );
 }
-function Child({ counter } = { counter: number }) {
-  <Child2 counter={counter} />;
+
+function Multiplications({ counter }) {
+  const [count, setCount] = counter;
+  return (
+    <div>
+      <p>Multiply count: {count}</p>
+      <MultBy5 setCount={setCount} />
+    </div>
+  );
 }
-function ExcessivePropDrilling() {
-  return <Child counter={5} />;
+
+function DivideBy5({ setCount }) {
+  return (
+    <button onClick={() => setCount((old) => old / 5)}>Divide by 5</button>
+  );
+}
+
+function Divisions({ counter }) {
+  const [count, setCount] = counter;
+  return (
+    <div>
+      <p>Divide count: {count}</p>
+      <DivideBy5 setCount={setCount} />
+    </div>
+  );
```

Comment:
What is this all about?

You should not pass a setfunction like this

Reply:
This is the wrong version where the person fixing should change these components to use composition instead of passing the prop down 👇🏽

The prop drilling example wasn't really doing anything with the prop, so fixing it was just a matter of removing the extra components